client/vault.go

```diff
@@ -48,7 +49,19 @@ func NewVaultAPI() (*APIClient, error) {
 	}
 	headers["X-Vault-Token"] = token
 
-	return NewAPIClient("vault", addr, headers), nil
+	tlsConfig, err := NewVaultTLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	apiClient := NewAPIClient("vault", addr, headers)
+
+	err = configureHttpClientTLS(apiClient.http.(*http.Client), tlsConfig)
```

Comment:
Agreed on your plan to pass the `TLSConfig` into `NewAPIClient()` on instantiation rather than mutating it by reference after. This would be a blocking review item if not!